storage/txcache/txByHashMap.go

```diff
@@ -0,0 +1,66 @@
+package txcache
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+)
+
+// TxByHashMap is
+type TxByHashMap struct {
+	Map     *ConcurrentMap
+	Counter core.AtomicCounter
+}
+
+// NewTxByHashMap creates a new map-like structure for holding and accessing transactions by txHash
+func NewTxByHashMap(size int, shardsHint int) TxByHashMap {
```

Comment:
this should get the concurrentmap as an input parameter.

Reply:
Actually, `TxByHashMap` wants to have full control over the map (and its associated counter). It wants to construct it by itself. But I have made it private and renamed it to "backingMap". Is it ok?